File: app/src/main/java/com/booyue/annotation/Unique.java
package com.booyue.annotation;

/**
 * Created by Administrator on 2018/4/26.09:29
 */

import android.support.annotation.StringDef;

import com.booyue.serial.SerialNumberManager;

@StringDef({SerialNumberManager.MAC,SerialNumberManager.SN})
public @interface Unique {
}

File: app/src/main/java/com/booyue/annotation/Unique.kt
package com.booyue.annotation

import android.support.annotation.StringDef
import com.booyue.serial.SerialNumberManager

/**
 * Created by Tianluhua on 2018\10\26 0026.
 */

@StringDef(SerialNumberManager.MAC, SerialNumberManager.SN)
annotation class Unique {
}